newa/__init__.py

```diff
@@ -719,7 +730,131 @@ class IssueAction:  # type: ignore[no-untyped-def]
 
 
 @define
-class ErratumConfig(Serializable):
+class ErratumConfig(Serializable):  # type: ignore[no-untyped-def]
+
+    project: str = field()
+    transitions: dict[str, list[str]] = field()
     issues: list[IssueAction] = field(  # type: ignore[var-annotated]
         factory=list, converter=lambda issues: [
             IssueAction(**issue) for issue in issues])
+
+
+@frozen
+class IssueHandler:
+    """ An interface to Jira instance """
+
+    url: str = field()
+    token: str = field()
+    project: str = field()
+
+    # Each project can have different semantics of issue status.
+    transitions: dict[str, list[str]] = field()
+
+    # We assume that all projects have the following two custom fields mapped
+    # as follows.
+    custom_field_map: ClassVar[dict[str, str]] = {
+        "field_epic_link": "customfield_12311140",
+        "field_epic_name": "customfield_12311141",
+        }
+
+    _connection: jira.JIRA = field(init=False)
+
+    @_connection.default  # pyright: ignore [reportAttributeAccessIssue]
+    def _connection_factory(self) -> jira.JIRA:
+        return jira.JIRA(self.url, token_auth=self.token)
+
+    def get_details(self, issue: Issue) -> jira.Issue:
+        """ Return issue details """
+
+        try:
+            return self._connection.issue(issue.id)
+        except jira.JIRAError as e:
+            raise Exception(f"Jira issue {issue} not found!") from e
+
+    def search_open_issues(self, query: str) -> tuple[jira.client.ResultList[jira.Issue], str]:
+        """ Search for issues that are not closed (returns subset of fields, only) """
+
+        fields = ["summary", "description", "status", "resolution"]
+
+        query = \
+            f"({query}) AND " + \
+            f"project = '{self.project}' AND " + \
+            f"status not in ({','.join(self.transitions["closed"])})"
```

Comment:
When new issues have `newa` label, we should include it in the query, right?

Reply:
Right, right, to be honest I am actually even thinking about moving all the search logic (including the stuff around identifier into the IssueHandler itself and creating an association between IssueHandler and ErratumJob. We only need to work with newa issue identifiers when working with Jira and hence this approach would reduce the code in jira command and make it a bit more readable. It would be a separate commit so that we can pick either version. That is most likely the last thing I have on my mind (it can be also done separately).

Reply:
Now IssueHandler is very generic and I don't thing that is needed. We only use it to map IssueActions to jira and work with them - all the actions are using newa identifier and so on. So it kinda make more sense to move the identifier logic into it and provide methods for search all issues related to specific errata job (all respins or current one or the old ones), we current we this code in jira command.

Reply:
Right, I think I rather meant adding it to the query which is later passed here as an argument.

Reply:
Sure, I'll add that. Previous comments were just a heads-up that it will be a part of bigger update. I'll get to that later today and that should conclude this PR.

Reply:
Done, I added one commit to move all Jira logic to IssueHandler, it made jira command implementation way more readable.